config/x_ac_zeromq.m4

```diff
@@ -4,5 +4,32 @@ AC_DEFUN([X_AC_ZEROMQ], [
 
     PKG_CHECK_MODULES([ZMQ], [libzmq >= 4.0.4])
 
+    old_CFLAGS=$CFLAGS
+    CFLAGS="$CFLAGS $ZMQ_CFLAGS"
+
+    old_LIBS=$LIBS
+    LIBS="$LIBS $ZMQ_LIBS"
+
+    AC_MSG_CHECKING([whether zeromq has CURVE support])
+    AC_RUN_IFELSE([
+        AC_LANG_SOURCE([
+            #include <zmq.h>
+	    int main () {
+	        char pub@<:@41@:>@;
+	        char sec@<:@41@:>@;
```

Comment:
some weird terminal cut & paste error?  I don't know why I'm seeing these funky characters.

Reply:
oh wait, length 41 char arrays.  why aren't the brackets showing up properly in my browser, it shows `@<:@`

Reply:
Those are m4 [quadrigraphs](https://www.gnu.org/software/autoconf/manual/autoconf-2.63/html_node/Quadrigraphs.html) (since you can't use `[]` inside `[]` in m4.)

Reply:
But I do wonder if "double quotes" `[[ ]]` would work here? I'm far from an m4 expert, but it _seemed_ to work in a simple test:
```diff
diff --git a/config/x_ac_zeromq.m4 b/config/x_ac_zeromq.m4
index 16b09fb0e..fdb974707 100644
--- a/config/x_ac_zeromq.m4
+++ b/config/x_ac_zeromq.m4
@@ -12,15 +12,15 @@ AC_DEFUN([X_AC_ZEROMQ], [
 
     AC_MSG_CHECKING([whether zeromq has CURVE support])
     AC_RUN_IFELSE([
-        AC_LANG_SOURCE([
+        AC_LANG_SOURCE([[
             #include <zmq.h>
            int main () {
-               char pub@<:@41@:>@;
-               char sec@<:@41@:>@;
+               char pub[41];
+               char sec[41];
                if (zmq_curve_keypair (pub, sec) < 0) return 1;
                return 0;
            }
-        ])],
+        ]])],
        [have_zmq_curve_support=yes; AC_MSG_RESULT([yes])],
        [AC_MSG_RESULT([no])]
     )
```

Reply:
thanks.  I thought it could be a special char format but couldn't figure out the special term to google.

Reply:
Nice!  [[  Is far better.